### ISSUE TITLE
Support "{ array: Position } as well as "[ Position ]" for array definitions

### ISSUE THREAD

**Issue by kitakun** [colyseus/schema#91]
This issue can be reproduced with https://github.com/colyseus/colyseus-unity3d/tree/release/0.14.0-alpha.1 branch and may be I Should create issue in another repository.
I have Schema like this:
```typescript
@type({ array: Position })
positions = new ArraySchema<Position>();
```
And after code generation I received this:
```csharp
[Type(0, "map", typeof(MapSchema<Position>))]
public MapSchema<Position> positions = new MapSchema<Position>();
```
Is this expected behaviour?

**Comment by endel** [comment]
Hi @kitakun, I'm glad you're trying out the latest alpha. 

Are you using the latest unity client as well? You can find it in the `dev` branch: https://github.com/colyseus/colyseus-unity3d/tree/dev ([Download link](https://github.com/colyseus/colyseus-unity3d/archive/dev.zip) - copy over just the "Assets/Plugins" folder)

Version 0.14 is mostly ready to be published (defold, haxe and cpp clients are still missing), if you can provide any feedback I really appreciate it.

**Comment by kitakun** [comment]
Hello again, and yes, I have reproduced this issue in tht last `dev` branch. I test all possible schemaCollections in example code:
![image](https://user-images.githubusercontent.com/3992943/93695135-93d4e000-fb3d-11ea-80f7-2e565777adf3.png)
![image](https://user-images.githubusercontent.com/3992943/93695141-a51dec80-fb3d-11ea-887b-7f4ae0959ee2.png)

**Comment by endel** [comment]
Hi @kitakun,

Sorry, I missed that you're `{ array: Position }` as type annotation, the correct supported way for arrays currently is `[Position]`:

```
@type([ Position ])
positions = new ArraySchema<Position>();
```

Indeed the other structures are not yet implemented in C# (set/collection)

Maybe it would make sense to allow using `{ array: Position }` to avoid confusion!

**Comment by endel** [comment]
Hi @kitakun, I've just published `@colyseus/schema@1.0.0-alpha.56` (on colyseus 0.14 alpha) fixing this on `schema-codegen`. You can now define your types as `@type({ array: XXX })` and have it properly generated as an array in the C# side. 🙏

**Comment by kitakun** [comment]
Thank you very much for this great news!
